package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -1115,13 +1141,18 @@ dnsmasq_start()
 	xappend "--dhcp-broadcast=tag:needs-broadcast"
 
 
-	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq.d"
+	# Create a dnsmasq.d dir for each instance
+	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
```

Comment:
Do you need to test it as being an absolute path, i.e. `"${dnsmasqconfdir:0:1}" = '/'` ?

Reply:
Just doing what original author did. Or did you have something like chroot or jail in mind?

Reply:
No, just trying to avoid ambiguity if someone thinks it’s a relative directory and it’s not.

Reply:
Could you give me an example that you think might be problematic?

Reply:
What if they have:

```
option confdir “..”
```

Reply:
OK. Got it. I suppose a user could do that. There does not seem to be harm in doing so, and that's possible today. The resulting path depends on how dnsmasq and the init script interpret it. Those will likely differ, and cause weird things to happen. 


We can test for absolute as your test. What are we thinking, gate all of this conf stuff behind e.g. this?
```sh
	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
	[ "${dnsmasqconfdir:0:1}" = '/' ] && {
	# Stuff
	}
```

Reply:
Looking at later lines, like `		dnsmasqconfdir="${dnsmasqconfdir%%,*}"` it seems that it maybe once upon a time accepted `dir1,dir2,dir3`.

Reply:
So, an iterator is needed? Or a `list` instead?

Reply:
I don't know. I think whoever wrote it gave up on the idea and settled on the first entry. Happy to let sleeping dogs lie...